plugins/obs-outputs/rtmp-stream.c

```diff
@@ -922,33 +961,41 @@ static inline bool send_headers(struct rtmp_stream *stream)
 
 	if (!send_audio_header(stream, i++, &next))
 		return false;
-
-	// send metadata only if HDR
-	video_t *video = obs_get_video();
-	const struct video_output_info *info = video_output_get_info(video);
-	enum video_colorspace colorspace = info->colorspace;
-	if (colorspace == VIDEO_CS_2100_PQ || colorspace == VIDEO_CS_2100_HLG)
-		if (!send_video_metadata(stream)) // Y2023 spec
-			return false;
-
-	if (!send_video_header(stream))
+	if (!send_video_metadata(stream, 0) || !send_video_header(stream, 0))
 		return false;
 
 	while (next) {
 		if (!send_audio_header(stream, i++, &next))
 			return false;
 	}
 
+	i = 1;
+	while (obs_output_get_video_encoder2(stream->output, i) != NULL) {
+		if (!send_video_metadata(stream, i) ||
+		    !send_video_header(stream, i))
+			return false;
+		i += 1;
+	}
+
 	return true;
 }
 
 static inline bool send_footers(struct rtmp_stream *stream)
 {
-	if (stream->video_codec == CODEC_H264)
-		return false;
+	for (size_t i = 0; i < MAX_OUTPUT_VIDEO_ENCODERS; i++) {
+		obs_encoder_t *encoder =
+			obs_output_get_video_encoder2(stream->output, i);
+		if (!encoder)
+			return true;
 
-	// Y2023 spec
-	return send_video_footer(stream);
+		if (stream->video_codec[i] == CODEC_H264)
+			continue;
```

Comment:
If H.264 is not on track 0, i.e. sent as "enhanced" packet, it should also send a footer.

Reply:
changed in https://github.com/obsproject/obs-studio/pull/10500#commits-pushed-bac01a2